Plugins/Format Source Code/plugin.swift

```diff
@@ -0,0 +1,93 @@
+import PackagePlugin
+import Foundation
+
+@main
+struct FormatSourceCode: CommandPlugin {
+  func performCommand(
+    context: PluginContext,
+    arguments: [String]
+  ) async throws {
+    let swiftFormatTool = try context.tool(named: "swift-format")
+    
+    // Extract the arguments that specify what targets to format.
+    var argExtractor = ArgumentExtractor(arguments)
+    let targetNames = argExtractor.extractOption(named: "target")
+    let targetsToFormat = try context.package.targets(named: targetNames)
+    
+    let configuration = argExtractor.extractOption(named: "configuration").first
+    
+    for target in targetsToFormat {
+      // Skip any type of target that doesn't have source files.
+      guard let target = target as? SourceModuleTarget else {
+        Diagnostics.warning("Skipping target “\(target.name)” because it doesn't contain Swift sources")
+        continue
+      }
+      
+      let swiftFormatExec = URL(fileURLWithPath: swiftFormatTool.path.string)
+      var swiftFormatArgs = [
+        "format",
+        target.directory.string,
+        "-r",
```

Comment:
Here and throughout, can you use the `--long-form` of the flags instead, so it's easier for a reader to understand when they quickly scan it?

Reply:
I don't know `--long-form`. what is this?

Reply:
Every flag that has a short form (like `-r`) has a longer form (`--recursive`). Using the longer ones will make the usage clearer. They're documented here: https://github.com/apple/swift-format#command-line-usage